The job **`tidy`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/16677301936/job/47207077063) [(plain enhanced)](https://triage.rust-lang.org/gha-logs/rust-lang/rust/47207077063) [(plain)](https://github.com/rust-lang/rust/commit/845c676ce220eed628bf63a97a880164f5233033/checks/47207077063/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
tidy check
Checking tidy rustdoc_json...
No error code explanation was removed!
tidy: Skipping binary file check, read-only filesystem
##[error]tidy error: /checkout/tests/ui/codegen/unsupported-static-initializer-in-const-array.rs:1: line longer than 100 chars
removing old virtual environment
creating virtual environment at '/checkout/obj/build/venv' using 'python3.10' and 'venv'
creating virtual environment at '/checkout/obj/build/venv' using 'python3.10' and 'virtualenv'
Requirement already satisfied: pip in ./build/venv/lib/python3.10/site-packages (25.1.1)
Collecting pip
---

added 291 packages in 33s
Running eslint on rustdoc JS files
info: ES-Check: there were no ES version matching errors!  🎉
some tidy checks failed
Command has failed. Rerun with -v to see more details.
Build completed unsuccessfully in 0:01:54
  local time: Fri Aug  1 14:19:03 UTC 2025
  network time: Fri, 01 Aug 2025 14:19:03 GMT
##[error]Process completed with exit code 1.
Post job cleanup.
```

</details>

--- a/tests/ui/codegen/unsupported-static-initializer-in-const-array.rs
+++ b/tests/ui/codegen/unsupported-static-initializer-in-const-array.rs
@@ -1,0 +1,18 @@
+//! LLVM error with unsupported expression in static
+//! initializer for const pointer in array on macOS.
+//!
+//! Regression test for <https://github.com/rust-lang/rust/issues/89225>.
+
+//@ build-pass
+//@ compile-flags: -C opt-level=3
+
+const fn make() -> (i32, i32, *const i32) {
+    const V: i32 = 123;
+    &V as *const i32;
+    (0, 0, &V)
+}
+
+fn main() {
+    let arr = [make(); 32];
+    println!("{}", arr[0].0);
+}

